imap_processing/ultra/l1b/ultra_l1b.py

```diff
@@ -0,0 +1,86 @@
+"""Calculates ULTRA L1b."""
+
+import json
+import logging
+import re
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+
+logger = logging.getLogger(__name__)
+
+
+def create_dataset(data_dict, name):
+    """
+    Create xarray for L1b data.
+
+    Parameters
+    ----------
+    data_dict: : dict
+        L1b data dictionary.
+
+    Returns
+    -------
+    dataset : xarray.Dataset
+        xarray.Dataset
+    """
+    # TODO: this will change to actual l1b data dictionary
+    # For now we are using it for retrieving the epoch.
+    dataset = data_dict["imap_ultra_l1a_45de"]
+
+    # Load metadata from the metadata file
+    with open(Path(__file__).parent.parent / "ultra_metadata_example.json") as f:
+        metadata = json.loads(f.read())[name]
+
+    epoch = dataset.coords["epoch"]
+
+    annotated_de_attrs = metadata["dataset_attrs"]
+
+    dataset = xr.Dataset(
+        coords={"epoch": epoch},
+        attrs=annotated_de_attrs,
+    )
+
+    dataset["x_front"] = xr.DataArray(
+        np.zeros(len(epoch), dtype=np.float32),
+        dims=["epoch"],
+        attrs=metadata["x_front"],
+    )
+
+    return dataset
+
+
+def ultra_l1b(data_dict: dict):
+    """
+    Process ULTRA L1A data into L1B CDF files at output_filepath.
+
+    Parameters
+    ----------
+    data_dict: : dict
+        Dictionary including the data itself and
+        its dependent data.
```

Comment:
There is an extra colon here and the dictionary is already implied by the type annotation
```suggestion
    data_dict : dict
        The data itself and its dependent data.
```